components/dashboard/src/settings/Integrations.tsx

```diff
@@ -199,77 +179,435 @@ function GitProviders() {
+            host,
+            type,
+            clientId,
+            clientSecret,
+            ownerId: props.userId
+        } as AuthProviderEntry.NewEntry : {
+            id: props.provider.id,
+            ownerId: props.userId,
+            clientId,
+            clientSecret,
+        } as AuthProviderEntry.UpdateEntry
+
+        setBusy(true);
+        setErrorMessage(undefined);
+        try {
+            const newProvider = await getGitpodService().server.updateOwnAuthProvider({ entry });
+
+            // wait two seconds for the changes to be propagated
+            await new Promise(resolve => setTimeout(resolve, 2000));
+
+            updateList();
+
+            // just open the authorization window.
+            /* await */ openAuthorizeWindow({ host: newProvider.host, onSuccess: updateList });
+
+            // close the modal, as the creation phase is done anyways.
+            close();
+        } catch (error) {
+            console.log(error);
+            setErrorMessage("message" in error ? error.message : "Failed to update Git provider");
+        }
+        setBusy(false);
+    }
+
+    const updateHostValue = (host: string) => {
+        if (props.mode === "new") {
+            setHost(host);
+            setRedirectURL(callbackUrl(host));
+            setErrorMessage(undefined);
+        }
+    }
+
+    const updateClientId = (value: string) => {
+        setClientId(value);
+    }
+    const updateClientSecret = (value: string) => {
+        setClientSecret(value);
+    }
+
+    const validate = () => {
+        const errors: string[] = [];
+        if (clientId.trim().length === 0) {
+            errors.push("Client ID is missing.");
+        }
+        if (clientSecret.trim().length === 0) {
+            errors.push("Client Secret is missing.");
+        }
+        if (errors.length === 0) {
+            setValidationError(undefined);
+            return true;
+        } else {
+            setValidationError(errors.join("\n"));
+            return false;
+        }
+    }
+
+    const getRedirectUrlDescription = (type: string, host: string) => {
+        let url = ``;
+        switch (type) {
+            case "GitHub":
+                url = `${host}/settings/developers`;
+                break;
+            case "GitLab":
+                url = `${host}/profile/applications`;
+                break;
+            default: return undefined;
+        }
+
+        return (<span>
+            Use this redirect URL to update the OAuth application.
+            Go to <a href={`https://${url}`} target="_blank" rel="noopener" className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">{url}</a> and setup the OAuth application.&nbsp;
+            <a href="https://www.gitpod.io/docs/gitlab-integration/#oauth-application" target="_blank" rel="noopener" className="text-gray-400 underline underline-thickness-thin underline-offset-small hover:text-gray-600">Learn more</a>.
```

Comment:
**issue**: Should this link to [GH OAuth Integration](https://www.gitpod.io/docs/github-integration/#oauth-application) docs page when selecting the GH provider type?